core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -1773,6 +1776,34 @@ protected SqlSelect createSourceSelectForDelete(SqlDelete call) {
     if (type != null) {
       return type;
     }
+    if (node instanceof SqlNamedParam) {
+      SqlNamedParam namedParam = (SqlNamedParam) node;
+      // Named Param is always in the default schema with the encoded table name.
+      String tableName = config().namedParamTableName();
+      if (tableName.equals(NAMED_PARAM_TABLE_NAME_EMPTY)) {
+        throw new RuntimeException("Named Parameter table is not registered. "
+            + "To use named parameters in a query please "
+            + "register a table name in the configuration.");
+      }
+      // TODO: Set caseSensitive?
+      CalciteSchema.TableEntry entry =
+          SqlValidatorUtil.getTableEntry(getCatalogReader(), ImmutableList.of(tableName));
+      if (entry == null) {
+        throw new RuntimeException("Named Parameter table is registered with "
```

Comment:
I think this should probably be a ValidationException. (org.apache.calcite.tools.ValidationException)

Reply:
Done